### ISSUE TITLE
Unreliable FFmpeg parsing code

### ISSUE THREAD

**Issue by desbma** [imageio/imageio#264]
I have a video that has several streams (video + audio + embedded image).

When I try to iterate on frames with:

    video = imageio.get_reader(filepath)
    for frame in range(video):
      # ...
The resulting frames are garbage images and I get :
`Warning: the frame size for reading (213, 120) is different from the source frame size (1916, 1076).`

I believe the problem comes from [that code](https://github.com/imageio/imageio/blob/29fe381dcb9ca16f973be95d93443247c172bd7d/imageio/plugins/ffmpeg.py#L497-L518) which does very unreliable parsing of `ffmpeg`'s output, and for this video file picks another stream's size as the "reading size".

Why not using `ffprobe` (part of FFmpeg) and and its structured output, that is easier to parse, more reliable, and would provide more readable code?

Example:

    import json, subprocess

    cmd = ("ffprobe", "-loglevel", "warning", "-of", "json", "-show_format", "-show_streams", filepath)
    output = subprocess.check_output(cmd)
    metadata = json.loads(output)
    first_video_stream = next(filter(lambda x: x["codec_type"] == "video", 
                                     metadata["streams"]))
    size = (first_video_stream["width"], first_video_stream["height"])

I can help implement the change if you agree.

**Comment by almarklein** [comment]
cc @kuchi 

We made some changes to the suspect code quite recently. Did you try with the latest version of imageio?

In principal I like this idea of using a more reliable way to get the meta-info on the video file, because we've struggled with the parsing of the ffmpeg output several times. However, this adds a dependency on `ffprobe`, which we do not provide at the moment. Perhaps we can add functionality to use `ffprobe`, and fallback to the current approach when its not available.

BTW: also see https://github.com/imageio/imageio/issues/115 where it is proposed to use the PyAV bindings to ffmpeg. That approach introduces even more dependencies (basically a dependency on conda), so I don't see that as a replacement of the current plugin.
